packages/commons/src/auth/jwk.ts

```diff
@@ -0,0 +1,33 @@
+import crypto, { JsonWebKey } from "crypto";
+import {
+  jwkDecodingError,
+  notAllowedPrivateKeyException,
+} from "pagopa-interop-models";
+
+export const decodeBase64ToPem = (base64String: string): string => {
+  try {
+    const cleanedBase64 = base64String.trim();
+    const decodedBytes = Buffer.from(cleanedBase64, "base64");
+    return decodedBytes.toString("utf-8");
+  } catch (error) {
+    throw jwkDecodingError(error);
+  }
+};
+
+// This method is to check if the key is public and to create the JWK If you don't add the if condition, createPublicKey turns any key into a public key
+export const createJWK = (pemKey: string): JsonWebKey => {
+  if (
+    pemKey.includes(
+      "-----BEGIN RSA PUBLIC KEY-----" || "-----BEGIN PUBLIC KEY-----"
+    )
```

Comment:
I was wondering to use the "crypto.createPrivateKey" function to check if the incoming key is private, instead of using this string check.
@galales what do you think?

Reply:
yes I agree, checking pem header is not enough to validate a public key.
Note that the crypto function accepts also a private key (because a public key can be derived from a private key).
You need to check the returned value to verify that the given pem was actually a public key (see documentation).

(@beetlecrunch I assume you were referring to [createPublicKey](https://nodejs.org/api/crypto.html#cryptocreatepublickeykey)).

Reply:
`createPublicKey` also works with the private key by deriving the public key.
I was thinking something like this
```
function isPublicKey(key) {
  try {
      crypto.createPrivateKey(key);
      return false;
  } catch (miss) {
      return true;
  }
}
```
I know it's bad, but it seems stronger than simple string control to me.

Reply:
the issue I see in this case is that we are validating that the passed pem is not a private key, and not that it is a public key.
Really dump example:
```
isPublicKey("blablabla")
```
returns `true`

`createPublicKey` should return info required to identify if the given parameter was a private or a public key

Reply:
Ok, I got you point.

> `createPublicKey` should return info required to identify if the given parameter was a private or a public key

Maybe I'm missing something, but `createPublicKey` returns a `KeyObject` type which cannot give us information about the type of the original key.

Reply:
I thought it could be told apart from the return value, but after double reading the docs it seems not.
First dirty solution that comes to mind is using both functions, where `createPublicKey` should succeed and `createPrivateKey` should fail

Reply:
Ok may I suggest this change? cc @galales 
```
function createPublicKey(key: string) {
  try {
    crypto.createPrivateKey(key);
    throw notAllowedPrivateKeyException();
  } catch {
    return crypto.createPublicKey(key);
  }
}

export const createJWK2 = (pemKey: string): JsonWebKey => {
  const publicKey = createPublicKey(pemKey);
  publicKey.export({ format: "jwk" });
}
```

Reply:
👌